applets/clock/clock.c

```diff
@@ -987,6 +996,80 @@ position_calendar_popup (ClockData *cd)
 
         gtk_window_move (GTK_WINDOW (cd->calendar_popup), x, y);
         gtk_window_set_gravity (GTK_WINDOW (cd->calendar_popup), gravity);
+    }
+#endif
+    /*Only build wayland support when building in process*/
+#ifdef CLOCK_INPROCESS
+#ifdef HAVE_WAYLAND
+
+    if (GDK_IS_WAYLAND_DISPLAY (gdk_display_get_default ()))
+    {
+        GtkWindow *window;
+        GdkWindow *panelwin;
+        GtkWidget *toplevel;
+        int  x, y, w, h, panel_w, panel_h;
+
+        /*Get the calendar window dimensions*/
+        window = (GTK_WINDOW (cd->calendar_popup));
+        gtk_window_get_size (window, &w, &h);
+        /*Find the position of the applet*/
+        gdk_window_get_origin (gtk_widget_get_window (cd->panel_button),
+                                                                      &x, &y);
+
+        /*Get the panel dimensions*/
+        toplevel = gtk_widget_get_toplevel (cd->applet);
+        panelwin = gtk_widget_get_window (toplevel);
+        gdk_window_get_geometry (panelwin, NULL, NULL, &panel_w, &panel_h);
+
+        /*Set up GTK Layer Shell*/
+        gtk_layer_init_for_window (window);
+        gtk_layer_set_layer (window, GTK_LAYER_SHELL_LAYER_TOP);
+
+        switch (cd->orient) {
+            case MATE_PANEL_APPLET_ORIENT_RIGHT:
+            gtk_layer_set_anchor (window,GTK_LAYER_SHELL_EDGE_LEFT, TRUE);
```

Comment:
style: space after comma (there are many of those)

Reply:
Done, I am assuming you mean use a space after each comma as in the rest of the file. That got missed probably on first cut and paste that generated one of these lines, then copied into all the others